tracing/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package opentelemetry
+
+import (
+	"context"
+	"io"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/thanos-io/objstore"
+)
+
+// TracingBucket is a wrapper around objstore.Bucket that adds tracing to all operations using OpenTelemetry.
+type TracingBucket struct {
+	tracer trace.Tracer
+	bkt    objstore.Bucket
+}
+
+func NewTracingBucket(tracer trace.Tracer, bkt objstore.Bucket) objstore.InstrumentedBucket {
```

Comment:
While we are here we could stop calling it "new" 🤔 . It's not new bucket. It's not even new bucket provider. Perhaps  `otel.TracedBucket(...)`or `WrapWithTracing`